eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -46,11 +62,13 @@ trait Eclair {
 
   def allupdates(nodeId: Option[PublicKey]): Future[Iterable[ChannelUpdate]]
 
-  def receive(description: String, amountMsat: Option[Long], expire: Option[Long]): Future[String]
+  def receive(description: String, amountMsat: Option[Long], expire: Option[Long], fallbackAddress: Option[String]): Future[String]
 
   def findRoute(targetNodeId: PublicKey, amountMsat: Long, assistedRoutes: Seq[Seq[PaymentRequest.ExtraHop]] = Seq.empty): Future[RouteResponse]
 
-  def send(recipientNodeId: PublicKey, amountMsat: Long, paymentHash: ByteVector32, assistedRoutes: Seq[Seq[PaymentRequest.ExtraHop]] = Seq.empty, minFinalCltvExpiry: Option[Long] = None): Future[PaymentResult]
+  def send(recipientNodeId: PublicKey, amountMsat: Long, paymentHash: ByteVector32, assistedRoutes: Seq[Seq[PaymentRequest.ExtraHop]] = Seq.empty, minFinalCltvExpiry: Option[Long] = None): Future[UUID]
+
+  def paymentInfo(id: UUID): Future[Option[OutgoingPayment]]
 
   def checkpayment(paymentHash: ByteVector32): Future[Boolean]
```

Comment:
```suggestion
  def receiveInfo(paymentHash: ByteVector32): Future[Boolean]
```

Reply:
It seems consistent to call it 'receiveInfo' however 'checkpayment' is really checking if the payment has been received and nothing else, it doesn't return generic information about it but just a boolean. I'd agree to call it 'receiveInfo' if we can do more than the bare check.

Reply:
This is not clear at all if `checkpayment` refers to an outgoing or incoming payment. Having it return an `Option[ReceivedPayment]` and be symmetrical with `sendInfo` is fine with me.